mne/channels/montage.py

```diff
@@ -679,20 +680,25 @@ def _backcompat_value(pos, ref_pos):
         not_in_montage = [name for name, use in zip(info_names, info_names_use)
                           if use not in ch_pos_use]
         if len(not_in_montage):  # DigMontage is subset of info
-            warn('DigMontage is a only a subset of info. '
-                 'There are %s channel position%s not present in '
-                 'the DigMontage. The required channels are: %s'
-                 % (len(not_in_montage), _pl(not_in_montage),
-                    not_in_montage))
-
-        # set ch coordinates and names from digmontage or nan coords
-        _ch_pos_use = dict()
-        for name, use in zip(info_names, info_names_use):
-            if use in ch_pos_use:
-                _ch_pos_use[use] = ch_pos_use[use]
-            else:
-                _ch_pos_use[use] = [np.nan, np.nan, np.nan]
-        ch_pos_use = _ch_pos_use
+            missing_coord_msg = 'DigMontage is a only a subset of info. ' \
+                                'There are %s channel position%s ' \
+                                'not present in the DigMontage. ' \
+                                'The required channels are: %s' \
+                                % (len(not_in_montage), _pl(not_in_montage),
+                                   not_in_montage)
+            if on_missing == 'raise':
+                raise ValueError(missing_coord_msg)
+            elif on_missing == 'ignore':
+                # set ch coordinates and names from digmontage or nan coords
+                _ch_pos_use = dict()
+                for name, use in zip(info_names, info_names_use):
+                    if use in ch_pos_use:
+                        _ch_pos_use[use] = ch_pos_use[use]
+                    else:
+                        _ch_pos_use[use] = [np.nan, np.nan, np.nan]
+                ch_pos_use = _ch_pos_use
+
+                warn(missing_coord_msg)
```

Comment:
This warning should be removed because if setting `on_missing='ignore'` users are already aware and explicitly opt in to this behavior.

Reply:
Agreed, probably most consistent to make it just a `logger.info`